### PR TITLE
Feat: S3 Object Lambda support

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*

None.

*Description of changes:*

Support for S3 Object Lambda - as described in this blog - https://aws.amazon.com/blogs/aws/introducing-amazon-s3-object-lambda-use-your-code-to-process-data-as-it-is-being-retrieved-from-s3/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by pankajagrawal16 — COMMENTED** [review]
🎉

**Comment by msailes** [comment]
Tested on AWS platform.

**Review by carlzogh — APPROVED** [review]

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/S3ObjectLambdaEvent.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+@Data
+@Builder(setterPrefix = "with")
+@NoArgsConstructor
+@AllArgsConstructor
+public class S3ObjectLambdaEvent {
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/S3ObjectLambdaEvent.java:14]: Could you please add a description of the event with a link to the relevant docs?

Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/S3ObjectLambdaEvent.java:20]: done

**Review by carlzogh — APPROVED** [review]

**Pull request merged**